packages/components/date-field/src/date-field.ts

```diff
@@ -0,0 +1,290 @@
+  @property({ type: Boolean }) override disabled?: boolean;
+
+  /**
+   * The first day of the week; 0 for Sunday, 1 for Monday.
+   * @default 1
+   */
+  @property({ type: Number, attribute: 'first-day-of-week' }) firstDayOfWeek?: number;
+
+  /** @internal Emits when the component gains focus. */
+  @event({ name: 'sl-focus' }) focusEvent!: EventEmitter<SlFocusEvent>;
+
+  /** @internal The input element in the light DOM. */
+  input!: HTMLInputElement;
+
+  /**
+   * The placeholder for the date field.
+   * @default undefined
+   */
+  @property() placeholder?: string;
+
+  /**
+   * Whether the date field is readonly.
+   * @default false
+   */
+  @property({ type: Boolean }) readonly?: boolean;
+
+  /**
+   * Whether the date field is a required field.
+   * @default false
+   */
+  @property({ type: Boolean, reflect: true }) override required?: boolean;
+
+  /**
+   * Whether the component is select only. This means you cannot type in the text field,
+   * but you can still pick a date via de popover.
+   * @default false
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'select-only' }) selectOnly?: boolean;
+
+  /**
+   * Shows the week numbers.
+   * @default false
+   */
+  @property({ type: Boolean, attribute: 'show-week-numbers' }) showWeekNumbers?: boolean;
+
+  /** The selected date in the calendar. */
+  @property({ converter: dateConverter }) override value?: Date;
+
+  /** @internal The wrapper element that is also the popover. */
+  @query('[part="wrapper"]') wrapper?: HTMLSlotElement;
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    if (!this.input) {
+      this.input = this.querySelector<HTMLInputElement>('input[slot="input"]') || document.createElement('input');
+      this.input.autocomplete = 'off';
+      this.input.slot = 'input';
+
+      if (!this.input.parentElement) {
+        this.append(this.input);
+      }
+    }
+
+    this.setFormControlElement(this.input);
+
+    // This is a workaround, because :has is not working in Safari and Firefox with :host element as it works in Chrome
+    const style = document.createElement('style');
+    style.innerHTML = `
+      sl-date-field:has(input:hover):not(:focus-within)::part(text-field) {
+        --_bg-opacity: var(--sl-opacity-light-interactive-plain-hover);
+      }
+    `;
+    this.prepend(style);
+  }
+
+  override willUpdate(changes: PropertyValues<this>): void {
+    super.willUpdate(changes);
+
+    if (changes.has('dateTimeFormat') && changes.has('locale')) {
+      this.#formatter = new Intl.DateTimeFormat(this.locale, this.dateTimeFormat);
+    }
+
+    if (changes.has('value')) {
+      console.log('value', this.value);
+
+      this.input.value = this.value && this.#formatter ? this.#formatter.format(this.value) : '';
+    }
+  }
+
+  override render(): TemplateResult {
+    return html`
+      <sl-text-field
+        @input=${this.#onInput}
```

Comment:
the typed in date in the text-field is not the same as shown in the calendar:

<img width="676" alt="image" src="https://github.com/user-attachments/assets/a64825a0-3df9-4499-a0d3-3edb39e4e109" />

So after typing in a proper date, the date in the calendar is not being updated.

Reply:
Yes, this is part of the component that is unfinished. But Bingel will only use `select-only`, so it's not necessary initially for them.